packages/react/src/components/Table/Table.story.jsx

```diff
@@ -2811,3 +2814,118 @@ export const RowExpansionAndLoadMore = () => {
 };
 
 RowExpansionAndLoadMore.storyName = 'row expansion: with load more ';
+
+export const WithColumnCustomizationModal = () => {
+  const demoGroupExample = boolean('demo grouping example', true);
+  const demoHasLoadMore = boolean('demo load more example (hasLoadMore)', true);
+  const pinnedColumnId = text('Pin column (pinnedColumnId)', 'string');
+  const hasVisibilityToggle = boolean('Allow toggling visibility (hasVisibilityToggle)', true);
+
+  const smallDataSet = tableData.slice(0, 5);
+  const allAvailableColumns = tableColumns;
+  const initialActiveColumns = allAvailableColumns.slice(0, 6);
+  const initialOrdering = [
+    { columnId: 'string' },
+    { columnId: 'date' },
+    { columnId: 'select' },
+    { columnId: 'secretField', isHidden: true },
+    { columnId: 'status' },
+    { columnId: 'number' },
+  ];
+
+  const columnGroupMapping = [
+    { id: 'groupA', name: 'Group A', columnIds: ['date', 'select'] },
+    { id: 'groupB', name: 'Group B', columnIds: ['status', 'secretField', 'number', 'boolean'] },
+  ];
+  const columnGroups = [
+    { id: 'groupA', name: 'Group A' },
+    { id: 'groupB', name: 'Group B' },
+  ];
+
+  const appendGrouping = (col) => {
+    const group = columnGroupMapping.find((group) => group.columnIds.includes(col.columnId));
+    return group
+      ? {
+          ...col,
+          columnGroupId: group.id,
+        }
+      : col;
+  };
+
+  const [showModal, setShowModal] = useState(true);
+  const [loadedColumns, setLoadedColumns] = useState(allAvailableColumns.slice(0, 7));
+  const [loadingMoreIds, setLoadingMoreIds] = useState([]);
+  const [canLoadMore, setCanLoadMore] = useState(true);
+  const [activeColumns, setActiveColumns] = useState(initialActiveColumns);
+  const [ordering, setOrdering] = useState(initialOrdering);
+  const [modalKey, setModalKey] = useState('initial-key');
+
+  return (
+    <>
+      <Table
```

Comment:
We should add the "MyTable" component based on the select knob like we've done in other stories, so we can test this against the StatefulTable, too.

Reply:
ignore. hadn't refreshed after you pushed this.

Reply:
Yea I was realising that when I was writing the acceptance test instructions